src/components/Disputes/DisputeAutoReveal.js

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { Button, GU, Info } from '@aragon/ui'
+import { useWallet } from '../../providers/Wallet'
+
+import { getAutoRevealRequest } from '../../services/autoReveal'
+import {
+  getAutoRevealPreference,
+  getCodeFromLocalStorage,
+  getOutcomeFromCommitment,
+} from '../../utils/crvoting-utils'
+
+function DisputeAutoReveal({ commitment, disputeId, onAutoReveal, roundId }) {
+  const { account } = useWallet()
+
+  const autoRevealRequested = useAutoRevealPolling(account, disputeId, roundId)
+
+  if (autoRevealRequested) {
+    return <Info>Auto reveal enabled!</Info>
+  }
+
+  return (
+    <RequestAutoReveal
+      commitment={commitment}
+      disputeId={disputeId}
+      onAutoReveal={onAutoReveal}
+      roundId={roundId}
+    />
+  )
+}
+
+function RequestAutoReveal({ commitment, disputeId, onAutoReveal, roundId }) {
+  const { account } = useWallet()
+  const autoRevealServiceEnabled = getAutoRevealPreference(account)
```

Comment:
I think the variable naming here is a bit confusing.

What do you think about renaming it to `autoRevealPreviouslyRequested`? And adding a comment that the service must have failed if we're rendering this component and it was previously requested?